packages/grid_client/src/client.ts

```diff
@@ -177,30 +158,38 @@ class GridClient {
 
   getDefaultUrls(network: NetworkEnv): Record<string, string> {
     const urls = { rmbProxy: "", substrate: "", graphql: "", activation: "", relay: "" };
-    if (network === NetworkEnv.dev) {
-      urls.rmbProxy = "https://gridproxy.dev.grid.tf";
-      urls.relay = "wss://relay.dev.grid.tf";
-      urls.substrate = "wss://tfchain.dev.grid.tf/ws";
-      urls.graphql = "https://graphql.dev.grid.tf/graphql";
-      urls.activation = "https://activation.dev.grid.tf/activation/activate";
-    } else if (network === NetworkEnv.test) {
-      urls.rmbProxy = "https://gridproxy.test.grid.tf";
-      urls.relay = "wss://relay.test.grid.tf";
-      urls.substrate = "wss://tfchain.test.grid.tf/ws";
-      urls.graphql = "https://graphql.test.grid.tf/graphql";
-      urls.activation = "https://activation.test.grid.tf/activation/activate";
-    } else if (network === NetworkEnv.qa) {
-      urls.rmbProxy = "https://gridproxy.qa.grid.tf";
-      urls.relay = "wss://relay.qa.grid.tf";
-      urls.substrate = "wss://tfchain.qa.grid.tf/ws";
-      urls.graphql = "https://graphql.qa.grid.tf/graphql";
-      urls.activation = "https://activation.qa.grid.tf/activation/activate";
-    } else if (network === NetworkEnv.main) {
-      urls.rmbProxy = "https://gridproxy.grid.tf";
-      urls.relay = "wss://relay.grid.tf";
-      urls.substrate = "wss://tfchain.grid.tf/ws";
-      urls.graphql = "https://graph.grid.tf/graphql";
-      urls.activation = "https://activation.grid.tf/activation/activate";
+    if (
+      !this.clientOptions.proxyURL ||
+      !this.clientOptions.relayURL ||
+      !this.clientOptions.substrateURL ||
+      !this.clientOptions.graphqlURL ||
+      !this.clientOptions.activationURL
+    ) {
```

Comment:
this means that I have to provide all the links, but we might change one of them and need the rest to be the default, you should use the value in the `clientOptions` (if any) or the default value.